svj_ntuple_processing/__init__.py

```diff
@@ -588,170 +577,163 @@ def veto_HEM(eta,phi,pt):
+    arrays.cutflow['triggers'] = len(a)
+    arrays.array = a
+    return arrays
 
-    # At least 2 AK15 jets
-    a = a[ak.count(a['JetsAK15.fCoordinates.fPt'], axis=-1) >= 2]
-    cutflow['n_ak15jets>=2'] = len(a)
+def select_trigger_object(arrays:Array):
+    # AK8 jetpt>500
+    a = arrays.array
+    a = a[ak.count(a['JetsAK8.fCoordinates.fPt'], axis=-1)>=1] # At least one jet
+    a = a[a['JetsAK8.fCoordinates.fPt'][:,0]>500.] # leading>500
+    arrays.cutflow['ak8jet.pt>500'] = len(a)
+    arrays.array = a
+    return arrays
 
-    # jetid for AK15 jets
-    a = a[a['JetsAK15_ID'][:,1]>0]
-    cutflow['ak15jets_id'] = len(a)
 
-    # subleading eta < 2.4 eta
-    a = a[np.abs(a['JetsAK15.fCoordinates.fEta'][:,1])<2.4]
-    cutflow['subl_eta<2.4'] = len(a)
-
-    if not single_muon_cr:
-        # Triggers
-        good_triggers, bad_triggers = copy.triggers
-        good_indices = np.array([copy.trigger_branch.index(t) for t in good_triggers])
-        bad_indices = np.array([copy.trigger_branch.index(t) for t in bad_triggers])
-        if len(a):
-            good_decisions = a['TriggerPass'].to_numpy()[:,good_indices]
-            a = a[(good_decisions == 1).any(axis=-1)]
-            # only keep events that fail the bad triggers
-            if len(bad_triggers):
-                bad_decisions = a['TriggerPass'].to_numpy()[:,bad_indices]
-                a = a[np.logical_not((bad_decisions == 1).any(axis=-1))]
-        cutflow['triggers'] = len(a)
-
-        # AK8 jetpt>500
-        a = a[ak.count(a['JetsAK8.fCoordinates.fPt'], axis=-1)>=1] # At least one jet
-        a = a[a['JetsAK8.fCoordinates.fPt'][:,0]>500.] # leading>500
-        cutflow['ak8jet.pt>500'] = len(a)
-
-    # positive ECF values
+def select_positive_ecf(arrays:Arrays):
     for ecf in [
         'JetsAK15_ecfC2b1', 'JetsAK15_ecfD2b1',
         'JetsAK15_ecfM2b1', 'JetsAK15_ecfN2b2',
         ]:
-        a = a[a[ecf][:,1]>0.]
-    cutflow['subl_ecf>0'] = len(a)
+        arrays.array = arrays.array[arrays.array[ecf][:,1]>0.]
+    arrays.cutflow['subl_ecf>0'] = len(arrays.array)
+    return arrays
+
 
+def select_rt(arrays: Arrays):
     # rtx>1.1
+    a = arrays.array
     rtx = np.sqrt(1. + a['MET'].to_numpy() / a['JetsAK15.fCoordinates.fPt'][:,1].to_numpy())
     a = a[rtx>1.1]
-    cutflow['rtx>1.1'] = len(a)
+    arrays.cutflow['rtx>1.1'] = len(a)
+    arrays.array = a
+    return arrays
 
-    # muon pt < 1500 filter to avoid highMET events
-    a = a[~ak.any(a['Muons.fCoordinates.fPt'] > 1500., axis=-1)]
-    cutflow['muonpt<1500'] = len(a)
-
-    if single_muon_cr:
-        # apply preselection - muon veto + muon selection
-        # (used medium ID + pt > 50 GeV + iso < 0.2 in EXO-19-020,
-        #  see AN-19-061 section 4.2)
-        # require the selected muon to match with the HLT muon object
-        # (which should be saved in the SingleMuon ntuples) by ΔR < 0.2
-        a = a[a['NMuons']>=1]
-        if len(a):
+
+def select_single_muon(copy Arrays):
```

Comment:
syntax error?

Reply:
Ah, This is just the git-diff showing odd values because some blocks has been reordered.

Reply:
I meant `copy Arrays` should be `copy: Arrays`